### ISSUE TITLE
No content displayed / no request is made

### ISSUE THREAD

**Issue by husjon** [bazukas/obs-linuxbrowser#110]
Hi

Using the latest `obs-linuxbrowser-bin` from AUR

```
╭─ husjon in ~ on (master)✘✘✘  
╰─(ﾉ˚Д˚)ﾉ  pikaur -Q obs-linuxbrowser-bin 
obs-linuxbrowser-bin 0.6.1-1
```

```
╭─ husjon in ~ on (master)✘✘✘  
╰─(ﾉ˚Д˚)ﾉ  pacman -Q obs-studio 
obs-studio 23.2.1-2
```

I'm not getting any content being displayed in the source.
![OBS Studio](https://i.imgur.com/YhIgpzs.png)

Using the default settings of `http://www.obsproject.com`, I'm expecting to see the content from that page, however nothing is shown.
![Linux Browser Settings](https://i.imgur.com/rmm0vDM.png)

I've tried letting it fetch content from one of my own servers, but no request is made.

If I've missed anything, please do let me know.

**Comment by 1stian** [comment]
Having the exact same issue.... Tried to reinstall CEF minimal and even tried to use the standard one available, no luck.

**Comment by NexAdn** [comment]
My first guess would be, that this is another occurence of #63 which I thought had been fixed. Is there CEF still installed despite you using the binary release? Also, does the problem occur, if you build obs-linuxbrowser from source (or install `obs-linuxbrowser` from AUR)?

**Comment by 1stian** [comment]
I've installed it with AUR. package name: obs-linuxbrowser-bin.
The package with only "obs-linuxbrowser" fails on make..(Error 1)

But i'll try from binary release as well.

**Comment by 1stian** [comment]
> My first guess would be, that this is another occurence of #63 which I thought had been fixed. Is there CEF still installed despite you using the binary release? Also, does the problem occur, if you build obs-linuxbrowser from source (or install `obs-linuxbrowser` from AUR)?

Hello, I am getting the same error by trying to build it from source...
`sudo make
[sudo] password for stian: 
[  9%] Building CXX object CMakeFiles/browser_shared.dir/src/browser/browser-app.cpp.o
In file included from /home/stian/Documents/obs-linuxbrowser-0.6.1/src/browser/browser-app.cpp:29:
/home/stian/Documents/obs-linuxbrowser-0.6.1/src/browser/browser-app.hpp:76:15: error: 'virtual bool BrowserApp::OnProcessMessageReceived(CefRefPtr<CefBrowser>, CefProcessId, CefRefPtr<CefProcessMessage>)' marked 'override', but does not override
   76 |  virtual bool OnProcessMessageReceived(CefRefPtr<CefBrowser> browser,
      |               ^~~~~~~~~~~~~~~~~~~~~~~~
/home/stian/Documents/obs-linuxbrowser-0.6.1/src/browser/browser-app.cpp: In member function 'virtual void BrowserApp::OnContextInitialized()':
/home/stian/Documents/obs-linuxbrowser-0.6.1/src/browser/browser-app.cpp:316:91: error: no matching function for call to 'CefBrowserHost::CreateBrowserSync(CefWindowInfo&, BrowserClient*, const char [45], CefBrowserSettings&, std::nullptr_t)'
  316 |      info, client.get(), "https://github.com/bazukas/obs-linuxbrowser/", settings, nullptr);
      |                                                                                           ^
In file included from /opt/cef/include/../include/cef_print_handler.h:42,
                 from /opt/cef/include/../include/cef_browser_process_handler.h:43,
                 from /opt/cef/include/cef_app.h:42,
                 from /home/stian/Documents/obs-linuxbrowser-0.6.1/src/browser/browser-app.hpp:23,
                 from /home/stian/Documents/obs-linuxbrowser-0.6.1/src/browser/browser-app.cpp:29:
/opt/cef/include/../include/cef_browser.h:305:32: note: candidate: 'static CefRefPtr<CefBrowser> CefBrowserHost::CreateBrowserSync(const CefWindowInfo&, CefRefPtr<CefClient>, const CefString&, const CefBrowserSettings&, CefRefPtr<CefDictionaryValue>, CefRefPtr<CefRequestContext>)'
  305 |   static CefRefPtr<CefBrowser> CreateBrowserSync(
      |                                ^~~~~~~~~~~~~~~~~
/opt/cef/include/../include/cef_browser.h:305:32: note:   candidate expects 6 arguments, 5 provided
/home/stian/Documents/obs-linuxbrowser-0.6.1/src/browser/browser-app.cpp: In member function 'void BrowserApp::UpdateActiveStateJS(bool)':
/home/stian/Documents/obs-linuxbrowser-0.6.1/src/browser/browser-app.cpp:387:17: error: 'class CefBrowser' has no member named 'SendProcessMessage'
  387 |  this->browser->SendProcessMessage(PID_BROWSER, msg);
      |                 ^~~~~~~~~~~~~~~~~~
/home/stian/Documents/obs-linuxbrowser-0.6.1/src/browser/browser-app.cpp: In member function 'void BrowserApp::UpdateVisibilityStateJS(bool)':
/home/stian/Documents/obs-linuxbrowser-0.6.1/src/browser/browser-app.cpp:395:17: error: 'class CefBrowser' has no member named 'SendProcessMessage'
  395 |  this->browser->SendProcessMessage(PID_BROWSER, msg);
      |                 ^~~~~~~~~~~~~~~~~~
make[2]: *** [CMakeFiles/browser_shared.dir/build.make:76: CMakeFiles/browser_shared.dir/src/browser/browser-app.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:73: CMakeFiles/browser_shared.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
`

**Comment by NexAdn** [comment]
Okay, so building it from source doesn't work with latest CEF versions due to #108. I'll have to fix that first, before we can check if it's related to the binary release or CEF in general.

**Comment by 1stian** [comment]
> Okay, so building it from source doesn't work with latest CEF versions due to #108. I'll have to fix that first, before we can check if it's related to the binary release or CEF in general.

OKaay. the "obs-linuxbrowser-bin" is not working for me at all... Do you got any ETA for it? This is the only part I am missing from my OBS.. I've just switched from Windows to Linux(Manjaro)..

**Comment by NexAdn** [comment]
Well, it seems they have changed the api (and even the version numbering) heavily and I haven't had time to investigate what exactly they changed, but tomorrow I can start looking into this.

**Comment by 1stian** [comment]
> Well, it seems they have changed the api (and even the version numbering) heavily and I haven't had time to investigate what exactly they changed, but tomorrow I can start looking into this.

Okay, awesome! But what version do you recommend to use in the mean while then? If I can find an earlier version of CEF

**Comment by NexAdn** [comment]
Earlier versions of CEF can be found on the official binary release download page (see README). In the past, version 3359 worked well. You can also try using the binary release without having CEF installed separetely.

**Comment by AlexWayfer** [comment]
Wow, it's breaking all things bug.

Sadly to experience it.

I hope somebody will have a time for making `obs-linuxbrowser` working again.

**Comment by AlexWayfer** [comment]
> Earlier versions of CEF can be found on the official binary release download page (see README). In the past, version 3359 worked well. You can also try using the binary release without having CEF installed separetely.

I confirm: the `68.0.3440.106` version of CEF, `master` version of `obs-linuxbrowser` and compiling by instructions in README on Arch Linux makes this plugin working. Thank you.

**Comment by AlexWayfer** [comment]
`minimal` builds work too.

`75.0.1+gb5e74dd+chromium-75.0.3770.80` works too.

`76.1.5+gd8a577c+chromium-76.0.3809.87` works.

Even the latest 64-bit `77.1.12+gc63c001+chromium-77.0.3865.90` works…

So, maybe the problem in the build?

**Comment by NexAdn** [comment]
There was a similar issue (#63) some time ago, causing only certain CEF builds to work with obs-linuxbrowser. To this day, however I have no clue why that might be, although it seemed to have been working for a while after merging #79. Apparently, this either was not the issue or there is some new issue with similar symptoms. Nonetheless, we most probably just have to keep track of the versions which are known to be working. I'll merge the `fix-new-cef-api-108-110` branch as soon as I have time testing these CEF builds myself.

**Comment by silmanduin66** [comment]
Has this problem been solved yet ? I tried it on archlinux but i got this error ```/usr//lib/obs-plugins/browser: symbol lookup error: /usr//lib/obs-plugins/browser: undefined symbol: cef_cookie_manager_create_manager```

**Comment by NexAdn** [comment]
This error is new to me. Have you updated CEF without recompiling obs-linuxbrowser?

**Comment by Finalgamer14** [comment]
> Has this problem been solved yet ? I tried it on archlinux but i got this error `/usr//lib/obs-plugins/browser: symbol lookup error: /usr//lib/obs-plugins/browser: undefined symbol: cef_cookie_manager_create_manager`

I'm having this same problem on manjaro. Tried installing cef-minimal from aur, and it's not working.

**Comment by FilipMadunicky** [comment]
I had the same problem when I installed the packages via AUR. 
I solved problem uninstalling AUR packages and then installing packages manually. I follow readme.md and download latest CEF from link and was follow steps. After build CEF I restart obs and it works.
